src/NHSD.BuyingCatalogue.Identity.Common/Extensions/StringExtensions.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+
+namespace NHSD.BuyingCatalogue.Identity.Common.Extensions
+{
+    public static class StringExtensions
+    {
+        public static string TrimAsync(this string input)
+        {
+            if (input is null)
+                throw new ArgumentNullException(nameof(input));
+
+            return input.Replace("async", "", StringComparison.OrdinalIgnoreCase);
```

Comment:
Rather than replace, shouldn't this just be stripping the Async off the end of the method name? I think there is an example in the `PasswordResetCallback` class, `Controller` method. Also I would be tempted to not throw an exception if the `input` is `null`. This means the method calling this extension doesn't have to check every time. If the input value is null just return null.

Reply:
As per our conversation, I agree with your point. After spending quite some time on making the compiler behave, I gave up and went with throwing if input is null. Given the use cases for this, it's not harmful to throw at that point (we would throw anyway, just bit later).

Reply:
👍